packages/edit-widgets/src/components/header/index.js

```diff
@@ -2,75 +2,96 @@
 import { PinnedItems } from '@wordpress/interface';
 import { useViewportMatch } from '@wordpress/compose';
+import { plus } from '@wordpress/icons';
+import { useRef } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import SaveButton from '../save-button';
-import useLastSelectedRootId from '../../hooks/use-last-selected-root-id';
 import UndoButton from './undo-redo/undo';
 import RedoButton from './undo-redo/redo';
-
-const inserterToggleProps = { isPrimary: true };
+import useLastSelectedRootId from '../../hooks/use-last-selected-root-id';
 
 function Header() {
+	const inserterButton = useRef();
 	const isLargeViewport = useViewportMatch( 'medium' );
-	const rootClientId = useLastSelectedRootId();
 	const isLastSelectedWidgetAreaOpen = useSelect(
 		( select ) =>
 			select( 'core/edit-widgets' ).getIsWidgetAreaOpen( rootClientId ),
 		[ rootClientId ]
 	);
-	const { setIsWidgetAreaOpen } = useDispatch( 'core/edit-widgets' );
+	const isInserterOpened = useSelect( ( select ) =>
+		select( 'core/edit-widgets' ).isInserterOpened()
+	);
+	const { setIsWidgetAreaOpen, setIsInserterOpened } = useDispatch(
+		'core/edit-widgets'
+	);
 	const { selectBlock } = useDispatch( 'core/block-editor' );
-
-	function handleInserterOpen( isOpen ) {
-		if ( isOpen && ! isLastSelectedWidgetAreaOpen ) {
-			// Select the last selected block if hasn't already.
-			selectBlock( rootClientId );
-			// Open the last selected widget area when opening the inserter.
-			setIsWidgetAreaOpen( rootClientId, isOpen );
-		}
-	}
+	const rootClientId = useLastSelectedRootId();
 
 	return (
 		<>
 			<div className="edit-widgets-header">
-				<NavigableToolbar
-					className="edit-widgets-header-toolbar"
-					aria-label={ __( 'Document tools' ) }
-				>
-					<ToolbarItem>
-						{ ( toolbarItemProps ) => (
-							<Inserter
-								position="bottom right"
-								showInserterHelpPanel
-								toggleProps={ {
-									...inserterToggleProps,
-									...toolbarItemProps,
-								} }
-								rootClientId={ rootClientId }
-								onToggle={ handleInserterOpen }
-							/>
-						) }
-					</ToolbarItem>
-					<ToolbarItem as={ UndoButton } />
-					<ToolbarItem as={ RedoButton } />
-					<ToolbarItem as={ BlockNavigationDropdown } />
-				</NavigableToolbar>
-				<h1 className="edit-widgets-header__title">
-					{ __( 'Block Areas' ) }
-				</h1>
+				<div className="edit-widgets-header__navigable-toolbar-wrapper">
+					<h1 className="edit-widgets-header__title">
+						{ __( 'Widgets' ) }
+					</h1>
+					<NavigableToolbar
+						className="edit-widgets-header-toolbar"
+						aria-label={ __( 'Document tools' ) }
+					>
+						<ToolbarItem
+							ref={ inserterButton }
+							as={ Button }
+							className="edit-widgets-header-toolbar__inserter-toggle"
+							isPrimary
+							isPressed={ isInserterOpened }
+							onMouseDown={ ( event ) => {
+								event.preventDefault();
+							} }
+							onClick={ () => {
```

Comment:
Nitpick: Let's move it to a function in the render function to make easier to read.